crates/sui-analytics-indexer/src/lib.rs

```diff
@@ -71,6 +72,27 @@ const DYNAMIC_FIELD_PREFIX: &str = "dynamic_field";
 
 const WRAPPED_OBJECT_PREFIX: &str = "wrapped_object";
 
+const ASYNC_TRANSACTIONS_TO_BUFFER_VAR_NAME: &str = "ASYNC_TRANSACTIONS_TO_BUFFER";
+const DEFAULT_ASYNC_TRANSACTIONS_TO_BUFFER: usize = 64;
+pub static ASYNC_TRANSACTIONS_TO_BUFFER: Lazy<usize> = Lazy::new(|| {
+    let async_transactions_opt = std::env::var(ASYNC_TRANSACTIONS_TO_BUFFER_VAR_NAME)
+        .ok()
+        .and_then(|s| s.parse().ok());
+    if let Some(async_transactions) = async_transactions_opt {
+        info!(
+            "Using custom value for '{}' max checkpoints in progress: {}",
+            ASYNC_TRANSACTIONS_TO_BUFFER_VAR_NAME, async_transactions
+        );
+        async_transactions
+    } else {
+        info!(
+            "Using default value for '{}' -- max checkpoints in progress: {}",
+            ASYNC_TRANSACTIONS_TO_BUFFER_VAR_NAME, DEFAULT_ASYNC_TRANSACTIONS_TO_BUFFER
+        );
+        DEFAULT_ASYNC_TRANSACTIONS_TO_BUFFER
+    }
+});
```

Comment:
Is this something that we may be able to put into a config somewhere vs an env var? Its fine if you think it would be too much work.